pkg/solana/txm/pendingtx.go

```diff
@@ -66,11 +70,21 @@ type finishedTx struct {
 	state       TxState
 }
 
+type txInfo struct {
+	id    string
+	state TxState
+}
+
+type retryCtx struct {
```

Comment:
Heads up here. Changed `cancelBy` to `sigToRetryCtx`. The idea is to be able to retrieve the `retryCtx` if a reorg was detected and only restart the retry/bumping cycle if the original `retryCtx` is still alive

Reply:
At least from my initial look, I thought we could keep this the same. If the retry context is still alive, that should mean the retry loop is still running so we probably don't need to do anything in that case. But if a confirmed transaction is re-org'd, we would have cancelled the retry context during state transition so that context would no longer be alive. We would still want to restart the retry loop for it though.